### PR TITLE
Issue 175: Updated build and install instructions

### PR DESCRIPTION
**Pull request by jdgarciauc3m**
Install and build instructions have been updated now.

### CODE REVIEW

**Review by Drioastorga — DISMISSED** [review]
I think that now the instructions are clear.

File: doc/install-notes.md [hunk 1]
```diff
@@ -1,98 +1,171 @@
...
 
-## Fully Supported Compilers ##
+## Building GrPPI
 
-  * **g++** from GNU. 
-  * **clang** from LLVM
+### Building the library
 
-  > Note: The versions of the compilers must be compatible with c++14
+GrPPI is a header only library and no build process is required for the library
```
Comment (mdolz) [inline doc/install-notes.md]: `header only` needs hyphen, so replace it by `header-only`.

File: doc/install-notes.md [hunk 2]
```diff
@@ -1,98 +1,171 @@
...
 
-  * [BOOST](http://www.boost.org/)
+* Building the unit tests.
+* Performing coverage analysis on unit tests.
+* Generating the **doxygen** based documentation.
+* Building the sample programs.
+* Installing the library in your system
 
-## Additional Libraries ##
+To setup the build scripts we recommend that you create an out of source
```
Comment (mdolz) [inline doc/install-notes.md]: `out of source` also needs hyphens, so replace it by `out-of-source`.

File: doc/install-notes.md [hunk 3]
```diff
@@ -1,98 +1,171 @@
...
+directory under the root GrPPI directory:
```
Comment (mdolz) [inline doc/install-notes.md]: Replace `root GrPPI directory` by `GrPPI root directory`.

File: doc/install-notes.md [hunk 4]
```diff
@@ -1,98 +1,171 @@
...
+### Disabling specific GrPPI backends
 
-  * [ccmake](https://cmake.org/cmake/help/v3.0/manual/ccmake.1.html)
-  	To enable/disable or modify options of the cmake configuration file through the terminal.
+You may want to disable specific back-ends. GrPPI offers specific variables to
+control this issue:
 
-  * [cmake-gui](https://cmake.org/cmake/help/v3.0/manual/cmake-gui.1.html)
-    Graphical tool to enable/disable or modify options of the cmake configuration file.
+* `GRPPI_OMP_ENABLE`: Enable/disable OpenMP backend.
```
Comment (mdolz) [inline doc/install-notes.md]: Replace `backend` by `back end`.

File: doc/install-notes.md [hunk 5]
```diff
@@ -1,98 +1,171 @@
...
+* `GRPPI_TBB_ENABLE`: Enable/disable TBB backend.
```
Comment (mdolz) [inline doc/install-notes.md]: Replace `backend` by `back end`.

File: doc/install-notes.md [hunk 6]
```diff
@@ -1,98 +1,171 @@
...
+~~~
+
+Then, you can build the unit tests by typing:
+
+~~~
+make
+~~~
 
-## Additional Programs ##
+### Disabling specific GrPPI backends
```
Comment (mdolz) [inline doc/install-notes.md]: Replace `backends` by `back ends`.

File: doc/install-notes.md [hunk 7]
```diff
@@ -1,98 +1,171 @@
...
+~~~
 
-## Using different compilers ##
+The coverage HTML reports are generated under `unit_tests/mycov/index.html`.
 
-1. The compiler must be selected before executing cmake in the build folder. If cmake has been executed and want to change the compiler, delete build folder or its contents.
+### Documentation generation
 
-2. To select the compiler is recommended to change the environement variable CXX, export CXX=/PATH_TO/COMPILER. Once defined the cmake can be executed. For example:
+Documentation generatio is disabled by default. However, if you wish to build
```
Comment (mdolz) [inline doc/install-notes.md]: Replace `generatio` by `generation`.

File: doc/install-notes.md [hunk 8]
```diff
@@ -1,98 +1,171 @@
...
+the documentation yourself, you may enable the option:
 
-	2.1. GNU:   export CXX=/usr/bin/g++
+~~~
+cmake .. -DGRPPI_DOXY_ENABLE=ON
+make
+~~~
 
-	2.2. Clang: export CXX=/opt/clang/bin/clang++
+This will generate a doc directory under your build tree with the generated
```
Comment (mdolz) [inline doc/install-notes.md:95]: Replace `with the generated documentation` by `with the documentation`.

File: doc/install-notes.md [hunk 9]
```diff
@@ -1,98 +1,171 @@
...
+## Required Libraries ##
+
+Miminal support of GrPPI requires the following libraries.
+
+  * [BOOST](http://www.boost.org/) version 1.58 or above.
 
-This tests include also the **coverage** option. Once the cmake has been executed, is possible to start the coverage executing make coverage_test-Name. At the end of the execution a new web browser window will be open with the coverage results. By default the browser selected is Mozilla Firefox. For example:
+## Additional Libraries ##
+
+In order to use the **Threading Building Blocks** (TBB) back-end you need to
```
Comment (mdolz) [inline doc/install-notes.md]: Shouldn't you add the word `Intel` right before `TBB` ?

File: doc/install-notes.md [hunk 10]
```diff
@@ -1,98 +1,171 @@
...
+install the following library:
 
-	make coverage_farm1_GT 
+  * [TBB](https://www.threadingbuildingblocks.org/)
+
+## Unit tests and coverage analysis
+
+For unit testing GrPPI uses the GoogleTest framework. However you do not need to
+install yourself. The framework is locally downloaded and compiled in your build
```
Comment (mdolz) [inline doc/install-notes.md]: Replace `install` by `install it`.

File: doc/install-notes.md [hunk 11]
```diff
@@ -1,98 +1,171 @@
...
-   		 cmake -DCMAKE_INSTALL_PREFIX=path/to/folder
+~~~
+ctest
+~~~
 
-3. To install execute **make install**. The dafault folder is /usr/local on UNIX and c:/Program Files on Windows.
+### Performing coverage analysis
 
-### Advanced Options
+To performe a coverage analysis type:
```
Comment (mdolz) [inline doc/install-notes.md]: Replace `performe` by `perform`.

**Review by mdolz — CHANGES_REQUESTED** [review]
See comments.

File: doc/install-notes.md (hunk 1 above)
Comment (mdolz) [reply doc/install-notes.md]: Replace also `and no build process` by `, so no build process`, or similar.

File: doc/install-notes.md (hunk 9 above)
Comment (mdolz) [reply doc/install-notes.md]: Replace `back-end` by `back end`.

File: doc/install-notes.md (hunk 6 above)
Comment (mdolz) [reply doc/install-notes.md]: See https://ell.stackexchange.com/a/117406 for more info on this.

**Review by javidoc00 — DISMISSED** [review]

File: doc/install-notes.md (hunk 8 above)
Comment (jdgarciauc3m) [reply doc/install-notes.md:95]: I don't agree. I really wanted to say "with the generated documentation"

**Review by mdolz — APPROVED** [review]

**Comment by jdgarciauc3m** [comment]
Solves issue #175

**Pull request merged**